dcc-portal-ui/app/scripts/advanced/js/advanced.js

```diff
@@ -65,8 +65,34 @@ angular.module('icgc.advanced.controllers', [
 
       _locationFilterCache = _filterService.getCachedFiltersFactory();
 
-      var _isInAdvancedSearchCtrl = true;
+      _controller.donorSets = _.cloneDeep(SetService.getAllDonorSets());
+      _controller.geneSets = _.cloneDeep(SetService.getAllGeneSets());
+      _controller.mutationSets = _.cloneDeep(SetService.getAllMutationSets());
+
+      // to check if a set was previously selected and if its still in effect
+      const checkSetInFilter = (entity, entitySets) => {
+        let filters = _locationFilterCache.filters();
+        const setIdPrefix = 'ES:';
+
+        if(filters[entity] && filters[entity].id){
+          _.each(entitySets, (set) => {
+            set.selected = false;
+            _.each(filters[entity].id.is, (id) => {
+              if(_.includes(id, setIdPrefix)){
+                if(`${setIdPrefix}${set.id}` === id){
+                  set.selected = true;
+                }
+              }
+            });
+          });
+        } else {
+          _.each(entitySets, (set) => {
+            set.selected = false;
+          });
+        }
+      };
```

Comment:
A "mutation" is changing an object

```js
var a = {
  foo: 1
};
a.foo = 2; // <-- mutation!
```

Take a quick glimpse at the wiki
https://en.wikipedia.org/wiki/Pure_function

I'd also recommend this article by Eric Elliot
https://medium.com/javascript-scene/master-the-javascript-interview-what-is-a-pure-function-d1c076bec976#.am312gdql

Naming things is one of the most important and hardest things we do, since it allows people to make assumptions of what a variable or function is or does before they read through all the code that it touches

We don't need to go full-functional and avoid mutations altogether, but when a function does perform mutations, we should take care not to give it a name that implies that it's a pure function

Functions with names that start with `get`, `is`, `check` would imply that it's something that returns a boolean and performs no mutations (that is not a exhaustive list, have to use your own judgement too)